### PR TITLE
Add Itunes command

### LINKED ISSUE #1322: Add iTunes commands.
I found an iTunes API open to use, and including search to some things related to apple like songs, movies, podcasts and more!

[Here api docs :)](https://affiliate.itunes.apple.com/resources/documentation/itunes-store-web-service-search-api/)

### PR DESCRIPTION
**Pull request by ghost**
Added an itunes command 

Closes #1322 

Edit: I am making this PR draft for now, As I am making the command better. I missed out some test cases and currently working.

### CODE REVIEW

**Comment by almeidx** [comment]
Please revert the changes you made to the env example file. If those are you bots Discord token and database uri, please make sure to change them

**Review by Doges — DISMISSED** [review]

**Review by almeidx — DISMISSED** [review]
Besides https://github.com/SwitchbladeBot/switchblade/pull/1345#discussion_r735606498, LGTM

**Comment by AsianCat54x** [comment]
Thanks @almeidx, I will fix that right now :)

**Comment by AsianCat54x** [comment]
Sorry, but forget to edit the config file in a hurry :(. Did the task, but the commit was from different account

File: src/commands/misc/itunes.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    super({
+      name: 'itunes',
+      requirements: { apis: ['itunes'] },
+      embedLogoURL: 'https://i.imgur.com/U4jjk5F.png',
+      parameters: [{
+        type: 'string',
+        full: false,
+        name: 'media',
+        whitelist: MEDIA_WHITE_LIST,
+        missingError: 'commands:searchplate.notFound',
```
Comment (almeidx) [inline src/commands/misc/itunes.js:16]: Wrong command

Comment (AsianCat54x) [reply src/commands/misc/itunes.js:16]: What do you mean? In the comment, it was told to use this command

Comment (almeidx) [reply src/commands/misc/itunes.js]: @Doges was just giving an example of how you can do it, you weren't supposed to use the same string!

File: src/locales/en-US/commands.json [hunk 2]
```diff
@@ -672,6 +672,21 @@
...
+  "itunes": {
+    "commandDescription": "Provides the content within the iTunes Store and Apple Books Store.",
+    "commandUsage": "<media> <term> <country code>",
+    "example": "s!itunes ebook Famous Five",
+    "place": "Itunes Store",
+    "title": "Showing Results for {{title}}.",
+    "track": "{{name}} - {{artistName}}",
+    "noMedia": "You will have to give me a valid type of media",
+    "availableMedia": "All the available media's are",
+    "ratingCount": "Rating Count: {{count}}",
```
Comment (almeidx) [inline src/locales/en-US/commands.json:684]: This is not being used
```suggestion
```

File: src/locales/en-US/commands.json [hunk 3]
```diff
@@ -672,6 +672,21 @@
...
   },
+  "itunes": {
+    "commandDescription": "Provides the content within the iTunes Store and Apple Books Store.",
+    "commandUsage": "<media> <term> <country code>",
+    "example": "s!itunes ebook Famous Five",
+    "place": "Itunes Store",
+    "title": "Showing Results for {{title}}.",
+    "track": "{{name}} - {{artistName}}",
+    "noMedia": "You will have to give me a valid type of media",
+    "availableMedia": "All the available media's are",
```
Comment (almeidx) [inline src/locales/en-US/commands.json:683]: This is not being used

```suggestion
```

File: src/locales/en-US/commands.json [hunk 4]
```diff
@@ -672,6 +672,21 @@
...
     "noText": "You have to give me some text to asciify!"
   },
+  "itunes": {
+    "commandDescription": "Provides the content within the iTunes Store and Apple Books Store.",
+    "commandUsage": "<media> <term> <country code>",
+    "example": "s!itunes ebook Famous Five",
+    "place": "Itunes Store",
+    "title": "Showing Results for {{title}}.",
+    "track": "{{name}} - {{artistName}}",
+    "noMedia": "You will have to give me a valid type of media",
```
Comment (almeidx) [inline src/locales/en-US/commands.json:682]: This is not being used

```suggestion
```

File: src/commands/misc/itunes.js (hunk 1 above)
Comment (AsianCat54x) [reply src/commands/misc/itunes.js]: Oh, my bad :((

File: src/locales/en-US/commands.json [hunk 5]
```diff
@@ -672,6 +672,21 @@
     "commandUsage": "<text>",
     "noText": "You have to give me some text to asciify!"
   },
+  "itunes": {
+    "commandDescription": "Provides the content within the iTunes Store and Apple Books Store.",
+    "commandUsage": "<media> <term> <country code>",
+    "example": "s!itunes ebook Famous Five",
```
Comment (almeidx) [inline src/locales/en-US/commands.json:678]: This is not being used

```suggestion
```

**Review by almeidx — CHANGES_REQUESTED** [review]

File: src/locales/en-US/commands.json [hunk 6]
```diff
@@ -672,6 +672,15 @@
     "commandUsage": "<text>",
     "noText": "You have to give me some text to asciify!"
   },
+  "itunes": {
+    "commandDescription": "Provides the content within the iTunes Store and Apple Books Store.",
+    "commandUsage": "<media> <term> <country code>",
+    "title": "Showing Results for {{title}}.",
+    "noMedia": "You will have to give me a valid type of media",
+    "availableMedia": "All the available media's are",
+    "noText": "You have to give me media and search terms!",
```
Comment (almeidx) [inline src/locales/en-US/commands.json:681]: Either use this in the missingError option in the second argument or remove this

File: src/commands/misc/itunes.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        full: false,
+        name: 'country',
+        required: false
+      }]
+    },
+    client)
+  }
+
+  async run (context, media, term, country = 'US') {
+    const data = await this.client.apis.itunes.search(media, term, country)
```
Comment (almeidx) [inline src/commands/misc/itunes.js:34]: Add an error message when nothing is found

File: src/locales/en-US/commands.json [hunk 8]
```diff
@@ -672,6 +672,11 @@
     "commandUsage": "<text>",
     "noText": "You have to give me some text to asciify!"
   },
+  "itunes": {
+    "commandDescription": "Provides the content within the iTunes Store and Apple Books Store.",
+    "commandUsage": "<media> <term> <country code>",
+    "notFound": "I wasn't able to find the term with the given media!'"
```
Comment (almeidx) [inline src/locales/en-US/commands.json:678]:
```suggestion
    "notFound": "I wasn't able to find the term with the given media!"
```

**Review by almeidx — CHANGES_REQUESTED** [review]

**Comment by almeidx** [comment]
CI is failing

**Review by almeidx — APPROVED** [review]

**Review by fernando457829 — DISMISSED** [review]
LGTM!

**Review by Doges — CHANGES_REQUESTED** [review]
Errors are not being catched
image

**Comment by AsianCat54x** [comment]
I didnt understand what are you trying to say

**Comment by Doges** [comment]
> 
> 
> I didnt understand what are you trying to say

Try to run the command with something like A Moment Apart, which is a song by odesza, it gives and error on console and does not send anything to the user

**Comment by AsianCat54x** [comment]
Oh oh oh, I understand. Gonna do right now

File: src/commands/misc/itunes.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+        required: true
+      }]
+    },
+    client)
+  }
+
+  async run (context, media, country, term) {
+    term = term.replaceAll(" " , "+")
+
+    const [data , response] = Array.from(await this.client.apis.itunes.search(media, term, country))
```
Comment (almeidx) [inline src/commands/misc/itunes.js:36]:
```suggestion
    const [data , response] = await this.client.apis.itunes.search(media, term, country)
```

File: src/commands/misc/itunes.js [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+  async parseResponse ({ channel, author, t }, data, title) {
+    const description = data.map((item, index) => `\`${String(index).padStart(2, '0')}\`: ${this.searchResultFormatter(item)}`)
+
+    const embed = new SwitchbladeEmbed(author)
+      .setThumbnail(this.embedUrl)
+      .setDescription(description)
+      .setColor(Constants.ITUNES_COLOR)
+      .setTitle(t('commands:itunes.title', { title }))
+
+    channel.send(embed)
```
Comment (almeidx) [inline src/commands/misc/itunes.js:68]:
```suggestion
    return embed
```

File: src/commands/misc/itunes.js [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+    return `[${i.trackName}](${i.trackViewUrl}) - [${i.artistName}](${i.artistViewUrl})`
+  }
+
+  getRatingEmojis (rating) {
+    return (this.getEmoji('ratingstar', '⭐').repeat(Math.floor(rating))) +
+            (this.getEmoji('ratinghalfstar')
+              .repeat(Math.ceil(rating - Math.floor(rating))))
+  }
+
+  async parseResponse ({ channel, author, t }, data, title) {
```
Comment (almeidx) [inline src/commands/misc/itunes.js:59]:
```suggestion
  async parseResponse ({ author, t }, data, title) {
```

File: src/commands/misc/itunes.js [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+
+    if(!response) {
+      throw new CommandError(context.t("commands:itunes.invalidTerm"))
+    }
+
+    if (data.length === 0) {
+      throw new CommandError(context.t('commands:itunes.noResults'))
+    }
+
+    this.parseResponse(context, data, context.message.content)
```
Comment (almeidx) [inline src/commands/misc/itunes.js:46]:
```suggestion
    channel.send(this.parseResponse(context, data, context.message.content))
```

File: src/commands/misc/itunes.js [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+      parameters: [{
+        type: 'string',
+        full: false,
+        name: 'media',
+        whitelist: MEDIA_WHITE_LIST,
+        missingError: 'commands:itunes.notFound',
+        required: true
+      }, {
+        type: 'string',
+        full: false,
```
Comment (almeidx) [inline src/commands/misc/itunes.js:20]:
```suggestion
```

**Review by almeidx — CHANGES_REQUESTED** [review]

**Review by almeidx — DISMISSED** [review]

**Review by moruzerinho6 — APPROVED** [review]

**Review by fernando457829 — APPROVED** [review]
LGTM!

**Review by Doges — APPROVED** [review]
LGTM

**Comment by Doges** [comment]
Just solve the conflicts before I merge

**Comment by AsianCat54x** [comment]
Oh okay :)

File: src/commands/misc/itunes.js [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+    return `[${i.trackName}](${i.trackViewUrl}) - [${i.artistName}](${i.artistViewUrl})`
+  }
+
+  getRatingEmojis (rating) {
+    return (this.getEmoji('ratingstar', '⭐').repeat(Math.floor(rating))) +
+            (this.getEmoji('ratinghalfstar')
+              .repeat(Math.ceil(rating - Math.floor(rating))))
+  }
+
+  async parseResponse (author, t, data, title) {
```
Comment (almeidx) [inline src/commands/misc/itunes.js:60]:
```suggestion
  parseResponse (author, t, data, title) {
```

File: src/commands/misc/itunes.js [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+
+    if (!response) {
+      throw new CommandError(t('commands:itunes.invalidTerm'))
+    }
+
+    if (data.length === 0) {
+      throw new CommandError(t('commands:itunes.noResults'))
+    }
+
+    channel.send(await this.parseResponse(author, t, data, message.content))
```
Comment (almeidx) [inline src/commands/misc/itunes.js:47]:
```suggestion
    channel.send(this.parseResponse(author, t, data, message.content))
```

**Review by almeidx — CHANGES_REQUESTED** [review]

**Review by almeidx — APPROVED** [review]

**Pull request closed without merging**